libindi/drivers/focuser/lacerta_mfoc.cpp

```diff
@@ -0,0 +1,649 @@
+{
+    INDI::Focuser::Disconnect();
+    return true;
+}
+
+/************************************************************************************
+ *
+************************************************************************************/
+const char *lacerta_mfoc::getDefaultName()
+{
+    LOGF_INFO("getDefaultName...", 0);
+
+    return (const char *)"Lacerta MFOC";
+}
+
+/************************************************************************************
+ *
+************************************************************************************/
+void lacerta_mfoc::ISGetProperties(const char *dev)
+{
+    LOGF_INFO("ISGetProperties...", 0);
+
+    if (dev != nullptr && strcmp(dev, getDeviceName()) != 0)
+        return;
+
+    INDI::Focuser::ISGetProperties(dev);
+
+    defineSwitch(&ModeSP);
+    loadConfig(true, "Mode");
+    
+    defineSwitch(&TempTrackDirSP);
+    loadConfig(true, "Temp. Track Direction");
+    
+    defineSwitch(&StartSavedPositionSP);
+    loadConfig(true, "Start at saved pos.");
+}
+
+/************************************************************************************
+ *
+************************************************************************************/
+bool lacerta_mfoc::initProperties()
+{
+    INDI::Focuser::initProperties();
+    LOGF_INFO("initProperties...", 0);
+
+    IUFillSwitch(&ModeS[MODE_ALL], "All", "All", ISS_ON);
+    IUFillSwitch(&ModeS[MODE_ABSOLUTE], "Absolute", "Absolute", ISS_OFF);
+    IUFillSwitch(&ModeS[MODE_RELATIVE], "Relative", "Relative", ISS_OFF);
+    IUFillSwitchVector(&ModeSP, ModeS, MODE_COUNT, getDeviceName(), "Mode", "Mode", MAIN_CONTROL_TAB, IP_RW,
+                       ISR_1OFMANY, 60, IPS_IDLE);
+
+    // Driver can define those to clients if there is support
+    IUFillNumber(&FocusAbsPosN[0], "FOCUS_ABSOLUTE_POSITION", "Ticks", "%4.0f", MFOC_POSMIN_HARDWARE, MFOC_POSMAX_HARDWARE, 10.0, 0);
+    IUFillNumberVector(&FocusAbsPosNP, FocusAbsPosN, 1, getDeviceName(), "ABS_FOCUS_POSITION", "Pos. Abs",
+                       MAIN_CONTROL_TAB, IP_RW, 60, IPS_OK);
+    
+    
+    IUFillNumber(&FocusRelPosN[0], "FOCUS_RELATIVE_POSITION", "Ticks", "%4.0f", 0.0, 100000.0, 10.0, 0);
+    IUFillNumberVector(&FocusRelPosNP, FocusRelPosN, 1, getDeviceName(), "REL_FOCUS_POSITION", "Rel. Pos.",
+                       MAIN_CONTROL_TAB, IP_RW, 60, IPS_OK);
+    
+    IUFillNumber(&BacklashN[0], "BACKLASH", "step", "%4.2f", 0, 255, 1, 12);
+    IUFillNumberVector(&BacklashNP, BacklashN, 1, getDeviceName(), "BACKLASH_SETTINGS", "Backlash", MAIN_CONTROL_TAB, IP_RW, 60,
+                       IPS_IDLE);
+    
+    IUFillNumber(&TempCompN[0], "TEMPCOMP", "step/10 degC", "%4.2f", -5000, 5000, 1, 65);
+    IUFillNumberVector(&TempCompNP, TempCompN, 1, getDeviceName(), "TEMPCOMP_SETTINGS", "T Comp.", MAIN_CONTROL_TAB, IP_RW, 60,
+                       IPS_IDLE);
+    
+    IUFillNumber(&PositionMaxN[0], "POSITIONMAX", "step", "%4.2f", MFOC_POSMIN_HARDWARE, MFOC_POSMAX_HARDWARE, 1, 8000);
```

Comment:
This is already init by the parent class. No need to do it here.

You need to override SetFocuserMaxPosition(...) to implement this functionality instead of defining a property.

Reply:
Sorry, the following is unclear to me : if removed, the driver GUI still features a position max. I can access to the value of variable, but how to call the update function ? I followed nfocus.cpp as reference, which seem to proceed in a similar way (MaxTravel). Which example would be more appropriate ?